.github/workflows/canary.yaml

```diff
@@ -0,0 +1,58 @@
+name: Canary Test
+
+on:
+  release:
+    types: [released]
+  schedule:
+  - cron: "30 18 * * *"
+  workflow_dispatch:
+  
+
+jobs:
+  upload:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    name: cache upload (${{ matrix.os }})
+    steps:
+    - name: Create file to cache
+      run: |
+        mkdir test-cache
+        echo "foo" > test-cache/test.txt
+    - uses: actions/cache@v3
+      with:
+        path: test-cache
+        key: ${{ matrix.os }}-runner-${{ github.run_number }}-${{ github.run_attempt }}
+  canary:
+    needs: upload
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    env:
+      CacheKey: ${{ matrix.os }}-runner-${{ github.run_number }}-${{ github.run_attempt }}
+    steps:
+      - name: checkout repo
+        uses: actions/checkout@v2
+      - name: Install extension
+        env:
+          GITHUB_TOKEN: ${{ secrets.DASHBOARD_TOKEN }}
+        run: gh extensions install actions/gh-actions-cache
+      - name: List Command
+        shell: bash
+        env:
+          GITHUB_TOKEN: ${{ secrets.REPO_WRITE_TOKEN }}
+        run: |
+          cache_found=$(gh actions-cache list | grep  $CacheKey)
+          echo $cache_found
+          [[ -z "$cache_found" ]] && exit 1 || echo "Cache List Successful"
+      - name: Delete Command
+        shell: bash
+        env:
+          GITHUB_TOKEN: ${{ secrets.REPO_WRITE_TOKEN }}
+        run: |
+          cache_delete=$(gh actions-cache delete $CacheKey --confirm | grep "Deleted 1 cache entry with key")
+          echo $cache_delete
+          [[ -z "$cache_delete" ]] && exit 1 || echo "Cache Delete Successful"
```

Comment:
Should we add slack notification if the workflow fails?
How will we come to know about failure in the current workflow?

Reply:
done